### PR TITLE
[Merged by Bors] - feat: `Prod.fst` as a lattice hom

### PR DESCRIPTION
**Pull request by YaelDillies**
From LeanCamCombi

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Order/Hom/Lattice.lean [hunk 1]
```diff
@@ -1586,6 +1586,48 @@ theorem symm_dual_comp (g : BoundedLatticeHom βᵒᵈ γᵒᵈ) (f : BoundedLat
...
+
+end LatticeHom
+
+/-! ### Pi -/
+
+namespace LatticeHom
+variable {ι : Type*} {α : ι → Type*} [∀ i, Lattice (α i)]
+
+/-- Evaluation as a lattice homomorphism. -/
+def evalLatticeHom (i : ι) : LatticeHom (∀ i, α i) (α i) where
```
Comment (urkud) [inline Mathlib/Order/Hom/Lattice.lean:1619]: Why repeat `LatticeHom` in NS and name?

Comment (YaelDillies) [reply Mathlib/Order/Hom/Lattice.lean:1619]: I think the naming is all over the place here

Comment (eric-wieser) [reply Mathlib/Order/Hom/Lattice.lean:1619]: I think we standardized on `FooHom.fst` and `Pi.evalFooHom`?

Comment (YaelDillies) [reply Mathlib/Order/Hom/Lattice.lean:1619]: Uh, really? That's kind of a terrible in-between

Comment (YaelDillies) [reply Mathlib/Order/Hom/Lattice.lean:1619]: Please just tell me what I should use and I will do it

Comment (eric-wieser) [reply Mathlib/Order/Hom/Lattice.lean:1619]: > Please just tell me what I should use and I will do it

Please rename the Prod version to `FooHom.fst`, as you had them before. The Pi version is correctly named as you have it right now.

Comment (eric-wieser) [reply Mathlib/Order/Hom/Lattice.lean:1619]: > Uh, really? That's kind of a terrible in-between

I think now that we're in Lean 4 we could reconsider the Pi naming in the face of dot notation (which works on the `.fst` definitions); but that's not something to mix into this PR.

**Review by eric-wieser — APPROVED** [review]
bors merge

Thanks!

**Pull request closed without merging**